src/visit.rs

```diff
@@ -350,7 +354,10 @@ impl<'ast> Visit<'ast> for DiscoveryVisitor<'_> {
         // statement referring to an external file. We remember the module
         // name and then later look for the file.
         if node.content.is_none() {
-            self.external_mods.push(mod_name.to_owned());
+            let mut mod_namespace = self.namespace_stack.clone();
```

Comment:
I noticed the existing code is using `namespace_stack` for both mods and functions. I guess it is harmless here because `mod` cannot occur inside functions...

It seems it would be good to at least clarify in the comment what goes onto the stack.

I'm not sure if we ever actually use the function namespace stack though. I'll look...

Reply:
I'll separate them; this PR is correct without it but it seems like it might cause confusion later.